internal/newusers/cache/db_test.go

```diff
@@ -309,7 +255,7 @@ func TestUserByID(t *testing.T) {
 		"Get existing user": {dbFile: "one_user_and_group"},
 
 		"Error on missing user":           {wantErrType: cache.NoDataFoundError{}},
-		"Error on invalid database entry": {dbFile: "invalid_entry_in_userByID", wantErrType: shouldError{}},
+		"Error on invalid database entry": {dbFile: "invalid_entry_in_userByID", wantErrType: cache.ErrNeedsClearing},
```

Comment:
I see this is checking that we're erroring, but shouldn't we also check the specific error for each case?

I mean, before we were checking that we had an error, and clearing, now we're checking whether the error is `cache.ErrNeedsClearing` but we don't ensure there's something else chained too, and if there's what it is.

Reply:
We don't want to assert on an external error type for safety reasons. What matters is that it returns an error and that this error also has the `ErrNeedsClearing` that triggers a cache cleaning.

The `ErrNeedsClearing` only gets joined in situations where we already had an error before, so the underlying error won't ever be nil.

Reply:
Well, that's in this case, but I assume also checking what is returned is relevant, isn't it?

Reply:
The returned errors are external ones (either from `bbolt` or `json`), so we shouldn't assert on them as they could change and break our tests. As long as an error is returned and it gets joined with `ErrNeedsCleaning`, our API did what it should've and we are good.